manage_breast_screening/notifications/management/commands/retry_failed_message_batch.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+import time
+from logging import getLogger
+
+from django.core.management.base import BaseCommand, CommandError
+
+from manage_breast_screening.notifications.management.commands.command_helpers import (
+    MessageBatchHelpers,
+)
+from manage_breast_screening.notifications.models import (
+    MessageBatch,
+    MessageBatchStatusChoices,
+)
+from manage_breast_screening.notifications.services.api_client import ApiClient
+from manage_breast_screening.notifications.services.queue import Queue
+
+logger = getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django Admin command which takes an ID of a MessageBatch with
+    a failed status and retries sending it to the Communications API.
+    """
+
+    def handle(self, *args, **options):
+        # get the batch from the queue in an iterator
+
+        queue = Queue.RetryMessageBatches()
+        failed_message_batch_iterator = queue.items(1)
+
+        # return if there isn't a message_batch
+        try:
+            failed_queue_message = failed_message_batch_iterator.next()
+        except StopIteration as e:
```

Comment:
Are we using a `try / except` block for flow control here?
I think the way we are retrieving 1 message from the queue is maybe a bit overly complex?
The `azure.storage.queue.QueueClient` implements `receive_message` for a single message, perhaps we can add and call a `Queue` method like `Queue.RetryMessageBatches().next_item()` to simplify and avoid flow control via exception blocks.